tests/integration/dump_test.go

```diff
@@ -603,3 +608,65 @@ func Test_Dump_KeysAndKeySets_Konnect(t *testing.T) {
 		})
 	}
 }
+
+// test scope:
+//
+// - >=3.10.0
+func Test_Dump_Deterministic_Sanitizer(t *testing.T) {
+	runWhen(t, "kong", ">=3.10.0")
+	setup(t)
+	// setup stage
+	client, err := getTestClient()
+	require.NoError(t, err)
+
+	ctx := context.Background()
+
+	tests := []struct {
+		name         string
+		stateFile    string
+		expectedFile string
+	}{
+		{
+			name:         "dump with sanitizer",
+			stateFile:    "testdata/dump/008-sanitizer/kong.yaml",
+			expectedFile: "testdata/dump/008-sanitizer/expected.yaml",
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			require.NoError(t, sync(context.Background(), tc.stateFile))
+
+			fileContent, err := file.GetContentFromFiles([]string{tc.stateFile}, false)
+			require.NoError(t, err)
+
+			sanitizer := sanitize.NewSanitizer(&sanitize.SanitizerOptions{
+				Ctx:     ctx,
+				Client:  client,
+				Content: fileContent.DeepCopy(),
+				Salt:    "deterministic-test-salt-12345",
+			})
+
+			sanitizedOutput, err := sanitizer.Sanitize()
```

Comment:
Why not `dump` with `--sanitize` flag - is it because the salt would be different and we'd end up getting different output?